### PR TITLE
python312Packages.pbr: fix python 3.12 compatibility; python312Packages.openstackdocstheme: allow for python 3.12; openstack for python 3.12

### LINKED ISSUE #362054: Build failure: python312Packages.jenkins-job-builder
## Steps To Reproduce

1. build `python312Packages.jenkins-job-builder`

## Build log

<details>

<summary>Build Log</summary>

```
error:
       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'shell'
         whose name attribute is located at /nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source/pkgs/stdenv/generic/make-derivation.nix:336:7

       … while evaluating attribute 'buildInputs' of derivation 'shell'
         at /nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source/pkgs/stdenv/generic/make-derivation.nix:383:7:
          382|       depsHostHost                = elemAt (elemAt dependencies 1) 0;
          383|       buildInputs                 = elemAt (elemAt dependencies 1) 1;
             |       ^
          384|       depsTargetTarget            = elemAt (elemAt dependencies 2) 0;

       … while evaluating the attribute 'out.outPath'
         at /nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source/lib/customisation.nix:352:13:
          351|             drvPath = assert condition; drv.${outputName}.drvPath;
          352|             outPath = assert condition; drv.${outputName}.outPath;
             |             ^
          353|           } //

       … while evaluating the attribute 'out.outPath'
         at /nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source/lib/customisation.nix:352:13:
          351|             drvPath = assert condition; drv.${outputName}.drvPath;
          352|             outPath = assert condition; drv.${outputName}.outPath;
             |             ^
          353|           } //

       … while calling the 'getAttr' builtin
         at <nix/derivation-internal.nix>:44:19:
           43|       value = commonAttrs // {
           44|         outPath = builtins.getAttr outputName strict;
             |                   ^
           45|         drvPath = strict.drvPath;

       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'python3.12-jenkins-job-builder-6.4.1'
         whose name attribute is located at /nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source/pkgs/stdenv/generic/make-derivation.nix:336:7

       … while evaluating attribute 'propagatedBuildInputs' of derivation 'python3.12-jenkins-job-builder-6.4.1'
         at /nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source/pkgs/stdenv/generic/make-derivation.nix:390:7:
          389|       depsHostHostPropagated      = elemAt (elemAt propagatedDependencies 1) 0;
          390|       propagatedBuildInputs       = elemAt (elemAt propagatedDependencies 1) 1;
             |       ^
          391|       depsTargetTargetPropagated  = elemAt (elemAt propagatedDependencies 2) 0;

       … while evaluating the attribute 'out.outPath'
         at /nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source/lib/customisation.nix:352:13:
          351|             drvPath = assert condition; drv.${outputName}.drvPath;
          352|             outPath = assert condition; drv.${outputName}.outPath;
             |             ^
          353|           } //

       … while calling the 'getAttr' builtin
         at <nix/derivation-internal.nix>:44:19:
           43|       value = commonAttrs // {
           44|         outPath = builtins.getAttr outputName strict;
             |                   ^
           45|         drvPath = strict.drvPath;

       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'python3.12-python-jenkins-1.8.2'
         whose name attribute is located at /nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source/pkgs/stdenv/generic/make-derivation.nix:336:7

       … while evaluating attribute 'nativeBuildInputs' of derivation 'python3.12-python-jenkins-1.8.2'
         at /nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source/pkgs/stdenv/generic/make-derivation.nix:380:7:
          379|       depsBuildBuild              = elemAt (elemAt dependencies 0) 0;
          380|       nativeBuildInputs           = elemAt (elemAt dependencies 0) 1;
             |       ^
          381|       depsBuildTarget             = elemAt (elemAt dependencies 0) 2;

       … while evaluating the attribute 'out.outPath'
         at /nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source/lib/customisation.nix:352:13:
          351|             drvPath = assert condition; drv.${outputName}.drvPath;
          352|             outPath = assert condition; drv.${outputName}.outPath;
             |             ^
          353|           } //

       … while calling the 'getAttr' builtin
         at <nix/derivation-internal.nix>:44:19:
           43|       value = commonAttrs // {
           44|         outPath = builtins.getAttr outputName strict;
             |                   ^
           45|         drvPath = strict.drvPath;

       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'python3.12-stestr-4.1.0'
         whose name attribute is located at /nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source/pkgs/stdenv/generic/make-derivation.nix:336:7

       … while evaluating attribute 'propagatedBuildInputs' of derivation 'python3.12-stestr-4.1.0'
         at /nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source/pkgs/stdenv/generic/make-derivation.nix:390:7:
          389|       depsHostHostPropagated      = elemAt (elemAt propagatedDependencies 1) 0;
          390|       propagatedBuildInputs       = elemAt (elemAt propagatedDependencies 1) 1;
             |       ^
          391|       depsTargetTargetPropagated  = elemAt (elemAt propagatedDependencies 2) 0;

       … while evaluating the attribute 'out.outPath'
         at /nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source/lib/customisation.nix:352:13:
          351|             drvPath = assert condition; drv.${outputName}.drvPath;
          352|             outPath = assert condition; drv.${outputName}.outPath;
             |             ^
          353|           } //

       … while calling the 'getAttr' builtin
         at <nix/derivation-internal.nix>:44:19:
           43|       value = commonAttrs // {
           44|         outPath = builtins.getAttr outputName strict;
             |                   ^
           45|         drvPath = strict.drvPath;

       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'python3.12-cliff-4.7.0'
         whose name attribute is located at /nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source/pkgs/stdenv/generic/make-derivation.nix:336:7

       … while evaluating attribute 'nativeBuildInputs' of derivation 'python3.12-cliff-4.7.0'
         at /nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source/pkgs/stdenv/generic/make-derivation.nix:380:7:
          379|       depsBuildBuild              = elemAt (elemAt dependencies 0) 0;
          380|       nativeBuildInputs           = elemAt (elemAt dependencies 0) 1;
             |       ^
          381|       depsBuildTarget             = elemAt (elemAt dependencies 0) 2;

       … in the condition of the assert statement
         at /nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source/lib/customisation.nix:352:23:
          351|             drvPath = assert condition; drv.${outputName}.drvPath;
          352|             outPath = assert condition; drv.${outputName}.outPath;
             |                       ^
          353|           } //

       … in the right operand of the IMPL (->) operator
         at /nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source/pkgs/development/interpreters/python/mk-python-derivation.nix:350:13:
          349| in extendDerivation
          350|   (disabled -> throw "${name} not supported for interpreter ${python.executable}")
             |             ^
          351|   { }

       … while calling the 'throw' builtin
         at /nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source/pkgs/development/interpreters/python/mk-python-derivation.nix:350:16:
          349| in extendDerivation
          350|   (disabled -> throw "${name} not supported for interpreter ${python.executable}")
             |                ^
          351|   { }

       error: openstackdocstheme-3.4.0 not supported for interpreter python3.12
```

</details>

## Additional context

Breaks options `services.jenkins.jobBuilder.*`

## Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.10.14, NixOS, 25.05 (Warbler), 25.05.20241203.55d15ad`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.25.2`
 - nixpkgs: `/nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source`

## Notify maintainers

@bot-wxt1221

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by SuperSandro2000**
Build python312Packages.jenkins-job-builder, python312Packages.jenkins-job-builder, python312Packages.python-cinderclient, python312Packages.python-openstack-client, python312Packages.python-openstackclient 

I forgot to mention that the reno change is required to fix python-cinderclient on python 3.12 otherwise it cannot be imported anymore. Not sure why exactly but since it was a none python package, probably some impurities where fixed somewhere.

Closes #362054

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Vinetos — APPROVED** [review]

**Review by bjornfor — COMMENTED** [review]
I made a fix for jenkins-job-builder that isn't a mass-rebuild (https://github.com/NixOS/nixpkgs/pull/363535), so it can be distributed to users more quickly (and also fixes jenkins-job-builder _runtime_ deps). Please rebase on top of that?

**Comment by SuperSandro2000** [comment]
The changes done in that PR fix the issue but in the wrong package and there are still some parts in pbr which then would be wrong. I won't accept them for openstackdocstheme in the long run and I really don't want to play the rebase game. Also the other dependency change is not quite correct.

**Comment by bjornfor** [comment]
@SuperSandro2000: I'm adding the backport label because we need to fix jenkins-job-builder on the NixOS release. I hope this PR was made with backporting in mind.

File: pkgs/development/python-modules/reno/default.nix [hunk 1]
```diff
@@ -79,6 +85,6 @@ python3Packages.buildPythonApplication rec {
     mainProgram = "reno";
     homepage = "https://docs.openstack.org/reno/latest";
     license = licenses.asl20;
-    maintainers = teams.openstack.members ++ (with maintainers; [ drewrisinger guillaumekoenig ]);
```
Comment (bjornfor) [inline pkgs/development/python-modules/reno/default.nix:82]: Why are the maintainers removed? IMHO, removal of maintainers requires more rationale than saying "misc cleanup" in the commit message.

**Pull request merged**